event_system.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+import json
+import os
+import time
+import multiprocessing as mp
+from dataclasses import dataclass, field, asdict, is_dataclass
+from typing import Any, Callable, Dict, List, Literal, Optional, Protocol, Set, Tuple
+
+class EventWriter:
+    def __init__(self, out_path: str):
+        self.out_path = out_path
+        os.makedirs(os.path.dirname(out_path), exist_ok=True)
+        open(self.out_path, "w").close()
```

Comment:
```suggestion
    def __init__(self, out_path: Path):
        self.out_path = out_path
		self.out_path.parent.mkdir(exist_ok=True, parents=True)
```

1. use `Path` from `pathlib`
2. ensure the file exists when you want to write to it, not before